src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -197,7 +192,11 @@ public string Comment
             set
             {
                 _fileComment = ZipHelper.GetEncodedTruncatedBytesFromString(value, _archive.EntryNameAndCommentEncoding, ushort.MaxValue, out bool isUTF8);
-                _hasUnicodeEntryNameOrComment |= isUTF8;
+
+                if (isUTF8)
+                {
+                    _generalPurposeBitFlag |= BitFlagValues.UnicodeFileNameAndComment;
```

Comment:
This is fine, since you're avoiding removing the bit flag unexpectedly if filename already set it to true.